Sources/ExtrasJSON/JSONValue.swift

```diff
@@ -217,3 +270,96 @@ public func == (lhs: JSONValue, rhs: JSONValue) -> Bool {
         return false
     }
 }
+
+extension JSONValue: ExpressibleByFloatLiteral {
+    public init(floatLiteral value: Double) {
+        self = .number(String(value))
+    }
+}
+
+extension JSONValue: ExpressibleByIntegerLiteral {
+    public init(integerLiteral value: Int) {
+        self = .number(String(value))
+    }
+}
+
+extension JSONValue: ExpressibleByStringLiteral {
+    public init(stringLiteral value: String) {
+        self = .string(value)
+    }
+}
+
+extension JSONValue: ExpressibleByBooleanLiteral {
+    public init(booleanLiteral value: Bool) {
+        self = .bool(value)
+    }
+}
+
+extension JSONValue: ExpressibleByArrayLiteral {
+    public init(arrayLiteral elements: JSONValue...) {
+        self = .array(elements)
+    }
+}
+
+extension JSONValue: ExpressibleByDictionaryLiteral {
+    public init(dictionaryLiteral elements: (String, JSONValue)...) {
+        self = .object(Dictionary(uniqueKeysWithValues: elements))
+    }
+}
+
+extension JSONValue: Codable {
+    public func encode(to encoder: Encoder) throws {
+        var container = encoder.singleValueContainer()
+        switch self {
+        case .number(let n):
+            if let int = self.intValue {
+                try container.encode(int)
+            } else if let double = self.doubleValue {
```

Comment:
Unfortunately, I think this might introduce rounding errors. Because JSON represents numbers as `String`, you might loose precision when converting to `Double` and then back to `String`. Unfortunately, I don't think there is an elegant way to avoid floating point arithmetic when going through the `Codable` interface.